### DISCUSSION TITLE
The field "syslog.facility.keyword" associated with this object no longer exists in the index pattern

### DISCUSSION THREAD

**Discussion by corycandia** [Security-Onion-Solutions/securityonion#5173] [Unsupported Versions]
Experts,

I am not getting anywhere with Google alone on this one.

(New-ish installation, 2.3.61, CentOS7, SO ISO Install, search manager + forward sensor)

I have two network devices sending syslog into SO, and VMware log insight forwarding into SO.  There are lots of syslogs.

I'm looking for not only the correct way to view the data (as opposed to trying to use artifacts left for people coming from migrated old versions), but I also keep getting this error:

![image](https://user-images.githubusercontent.com/88662234/129709454-8b92d32a-3ab4-4a37-9f5b-8a99d96abaf8.png)

When editing the visualization:
**Error**
"The field "syslog.facility.keyword" associated with this object no longer exists in the index pattern. Please use another field"
The field "syslog.severity.keyword" associated with this object no longer exists in the index pattern. Please use another field.
![image](https://user-images.githubusercontent.com/88662234/129709885-de71e01f-0ceb-4d38-99f5-c5021af3ed76.png)

I have already gone into the Index pattern once and simply added the two patterns once, looks like it was removed.  I don't yet fully understand how this is populated by SO or why it would be removed after a period of time other than if I need to edit a template somewhere.  

![image](https://user-images.githubusercontent.com/88662234/129709706-e71cf166-b837-49e0-aeb5-18f9630e4139.png)

**Questions**
1. Is anyone else's broke or just mine?  

2. I didn't change anything, this is just 'out of the box'. Is it expected that you customize something to establish syslog maybe I didn't do?

3. Am I viewing/searching them correctly?

I need to learn the difference between 'Module:syslog' and 'dataset:syslog'

4. Does it seem right that there is no data being populated in IPs and Ports?  Maybe that isn't supposed to be there?
![image](https://user-images.githubusercontent.com/88662234/129710640-6dd1a541-4385-45a2-87d7-e25069c0c3ec.png)

5. What would the best forwarded event format be for SO between 'syslog', 'ingestion api', and 'raw'?
![image](https://user-images.githubusercontent.com/88662234/129714718-78cdfcc8-27ec-43ee-b526-f8f607ca7680.png)

**Comment by weslambert** [comment]
If you look at a raw event by expanding the records in Kibana, is the syslog.facility field present?

**Reply by corycandia** [reply]
It (facility) looks like it is not, which is surprising.  Maybe that's why I'm confused, either it was in the previous version, or I assumed it would be and just surprised by this clarification.

After getting a crash course in E-Stack components, I'm figuring out how SO creates some of the pipeline objects with SALT rather than or in contrast to a standard Elastic solution that uses the GUI and ETC files.

I found a thread ([https://github.com/Security-Onion-Solutions/securityonion/discussions/3906](https://github.com/Security-Onion-Solutions/securityonion/discussions/3906)) where you recommended someone writing their own ingest and placing them where they are managed by SALT (I'm getting the impression that you guys are using SALT for literally everything).  Once I have figured that out, I may try to go further with this.

Just like in the case of that other discussion's OP (3906), I want to expand on the information coming into SYSLOG greatly.  Right now, most of the information I want is being lumped together in the message portion, like facility, severity, firewall items, web proxy information from Sophos, exchange email message tracking, etc.  It all gets piled into message and needs to be dissected or GROK'd out.  In one sense, I wonder if I'm blurring the main purpose of SO being an alert system and trying to ask it to do more general network management and reporting, but I think it easily can, just not out of the box.

@weslambert Thanks for your attention to the forums, even when the questions are so novice.
